capstone/render.md

```diff
@@ -0,0 +1,244 @@
+# Deploying on Render
+
+Deployment is a very exciting step in the development process. Deployment is the process of employing a service to store your source code and data on their servers and providing a pathway in the form of the url. Render is platform used to host web applications. Render's services support many languages and it offers 3 month free deployments for small projects.
+
+It is perfectly normal to deploy an application that is not "finished." Projects are always evolving and require regular deployment updates. The beauty of using a deployment service like Render is that you can connect the deployment to your GitHub repository. By default, Render will automatically build and deploy your services with every code push to your remote repository.
+
+### Prep List
+
+As a team, decide who will run the commands for deploying the application. Typically the tech lead would take on this responsibility.
+
+Prior to deployment we need to ensure the main branch is up to date and the project runs without errors.
+
+Next, if we have not already, we will create a Render acount using the team's email. Once created, need access to the email and password to log into the team's Render account.
+
+And finally, the person in charge of deploying the application will need access to the `master.key` credential that lives in the `config` folder. The key is used to encrypt and decrypt your credentials. The `master.key` is not checked in to version control and so will not exist on Github. If the project was cloned from GitHub there will not be a `master.key` file in the `config` folder. If the person deploying the application has a `master.key` file, great! You are ready to go. If not, continue with the following steps in terminal.
+
+```bash
+$ EDITOR="vim" bin/rails credentials:edit
+$ :wq
+```
+
+The first command will create the `master.key` file and a new `config/credentials.yml.enc` file with the Rails secrets. The command will drop you into a `vim` editor in your terminal. The second command will save the changes.
+
+While the `master.key` will not be stored in version control, the `credentials.yml.enc` will be. So be sure that any updates are pushed to GitHub.
+
+### Rails Application Configuration
+
+First we will need to find the `database.yml` file in our `config` directory and modify the production section to gather the database configuration from the `DATABASE_URL` environment variable. 
+
+**config/database.yml**
+
+```yaml
+ production:
+  <<: *default
+  url: <%= ENV['DATABASE_URL'] %>
+```
+Next, in our `puma.rb` file also found in our `config` directory, we are going to uncomment and update the following lines of code. 
+
+```ruby
+workers EVN.fetch("WEB_CONCURRENCY") {4}
+```
+changing the `{2}` to `{4}`
```

Comment:
One option: Change the `{2}` to `{4}`.

Second option: Change the `{2}` to `{4}` because of a reason that is listed here.